### PR TITLE
28 dev env docker

### PR DESCRIPTION
**Pull request by lazovik**

### CODE REVIEW

File: src/main/scala/rugds/sbt/DockerDevEnvironment.scala [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+              case (port, _) => s"-p $port:$port"
+            }.mkString(" ")
+            val dockerRunCmd = s"docker run -d --network=$networkId $exposedPorts --name ${docker.name} ${docker.toString}"; log.info(dockerRunCmd)
+            dockerRunCmd.!!
+          }
+          docker
+      }.toSet
+    },
+
+    envDockerList := (resources in Compile).value
```
Comment (BrianSetz) [inline src/main/scala/rugds/sbt/DockerDevEnvironment.scala]: Is this only for the docker dependencies inside the current project, or also for external?

File: src/main/scala/rugds/sbt/DockerDevEnvironment.scala [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    },
+
+    envStartDev := {
+      val log = streams.value.log
+      val pdi = envPullDockerImages.value
+//      val networkCmd = s"docker -P" // TODO: auto-create network if not yet created (or fail if exists?)
+      val networkId = "rugds-dev"
+      envDockerDependencyTree.value.map {
+        case (docker, dependencies) =>
+          val dockerPsCmd = Seq("docker", "ps", "-aq", "--format", "{{ .Names}}"); log.info(dockerPsCmd.mkString(" "))
```
Comment (BrianSetz) [inline src/main/scala/rugds/sbt/DockerDevEnvironment.scala]: --format vs -f

File: src/main/scala/rugds/sbt/DockerDevEnvironment.scala [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        repo.listFiles.map(image => {
+          val imageVersion = scala.io.Source.fromFile(image.getAbsolutePath + "/version").mkString
+          val dockerBuild  = if (new File(image.getAbsolutePath + "/Dockerfile").exists()) Some(image) else None
+          EnvDocker(repo.name, image.name, imageVersion, dockerBuild)
+        })
+      })
+      .filter(_.file.isDefined)
+      .toSet,
+
+    envDockerBuild := envDockerList.value
```
Comment (BrianSetz) [inline src/main/scala/rugds/sbt/DockerDevEnvironment.scala]: See above comment, does this only build images for current project, or also for external

File: src/main/scala/rugds/sbt/DockerDevEnvironment.scala (hunk 1 above)
Comment (lazovik) [inline src/main/scala/rugds/sbt/DockerDevEnvironment.scala]: without following dependencies (also, excluding subprojects)

**Pull request merged**